crates/pathfinder/src/sequencer.rs

```diff
@@ -474,23 +475,24 @@ impl ClientApi for Client {
         contract_address_salt: ContractAddressSalt,
         constructor_calldata: Vec<ConstructorParam>,
         contract_definition: ContractDefinition,
+        token: Option<String>,
     ) -> Result<reply::add_transaction::DeployResponse, SequencerError> {
         let req =
             request::add_transaction::AddTransaction::Deploy(request::add_transaction::Deploy {
                 contract_address_salt,
                 contract_definition,
                 constructor_calldata,
             });
+        let mut url = self.build_query(&["gateway", "add_transaction"], &[]);
+        // this is an optional token currently required on mainnet
+        if let Some(token) = token {
```

Comment:
should we allow this hidden arg to be available for testnet too?

Reply:
Yeah, I think it's more straightforward this way.
I don't think putting in more logic than what's strictly necessary makes sense here...